src/eos-updater-poll.c

```diff
@@ -203,6 +203,192 @@ get_fetchers (SourcesConfig *config,
 }
 
 static void
+object_unref0 (gpointer obj)
+{
+  if (obj == NULL)
+    return;
+  g_object_unref (obj);
+}
+
+static GPtrArray *
+get_finders (SourcesConfig          *config,
+             GMainContext           *context,
+             OstreeRepoFinderAvahi **out_finder_avahi)
+{
+  g_autoptr(OstreeRepoFinderAvahi) finder_avahi = NULL;
+  g_autoptr(GPtrArray) finders = g_ptr_array_new_full (config->download_order->len,
+                                                       object_unref0);
+  g_autoptr(GError) local_error = NULL;
+  gsize i;
+
+  g_assert (config->download_order->len > 0);
+
+  for (i = 0; i < config->download_order->len; i++)
+    {
+      switch (g_array_index (config->download_order, EosUpdaterDownloadSource, i))
+        {
+        case EOS_UPDATER_DOWNLOAD_MAIN:
+          g_ptr_array_add (finders, ostree_repo_finder_config_new ());
+          break;
+
+        case EOS_UPDATER_DOWNLOAD_LAN:
+          finder_avahi = ostree_repo_finder_avahi_new (context);
+          g_ptr_array_add (finders, g_object_ref (finder_avahi));
```

Comment:
I think `finder_avahi` will be leaked here if we have multiple lan sources - it makes more sense to me to just pass `ostree_repo_finder_avahi_new` directly to `g_ptr_array_add`

Reply:
Oh, hmm, I guess its only possible to have one `finder_avahi`. Maybe an assertion that it isn't already set?

Reply:
`strv_to_download_order()` checks that there are no duplicated sources, but I’ll add an assertion to make it clearer.

Reply:
Assertion added in a fixup.